### ISSUE TITLE
Modified gridworld

### ISSUE THREAD

**Issue by albertoxamin** [Unity-Technologies/ml-agents#3017]
I have an environment that is similar to gridworld, there is a grid of 4x4, the agent is the yellow square 🟨, the green cells are the ones that the agent has visited, and the red ones are the cells that have not been visited.  
![image](https://user-images.githubusercontent.com/6067659/70042045-87879280-15be-11ea-953f-0feeb415cca4.jpeg)
I want my agent to visit all the cells in the least amount of time.
So I tried to add a `-0.01` reward at every step and `+1+Done()` when the agent has visited all the cells.
The training (I'm using the same gridworld hyperparameters with SAC) mean reward keeps decreasing and after around 170k steps the agent stayed still and didn't complete any other training episode, since by standing still the agent never calls Done.
Do you have any idea why that might be the case and how could I achieve the desired behaviour?

**Comment by afewvowels** [comment]
I recently started training on SAC because I read how much more efficient it was and discovered the same behavior of inaction. I followed a tutorial on making a chameleon and came down in the morning to find 25 green chameleons which totally ignored their surroundings and the reward signals. They just immediately changed colors from the random start color to green and stayed there. Another simple test chamber that had an agent, a ball, and a goal resulted in the agent spinning in circles after 50k steps, despite having moved the ball into the goal several times earlier in the training.

I'm going back to PPO and having good results, I'd suggest you try the same.

**Comment by afewvowels** [comment]
.NET Framework 4.x does not result in working SAC agents, they still converge on a set of behaviors that does not maximize rewards but instead favors a single type of behavior. I got blue chameleons this time.

I'm going back to PPO.

These were my training parameters, and a link to the repo if you want to try to replicate the behavior yourself:

```yaml
default:
    trainer: sac
    batch_size: 128
    buffer_size: 50000
    buffer_init_steps: 0
    hidden_units: 128
    init_entcoef: 1.0
    learning_rate: 3.0e-4
    learning_rate_schedule: constant
    max_steps: 1.0e5
    memory_size: 256
    normalize: false
    num_update: 1
    train_interval: 1
    num_layers: 2
    time_horizon: 64
    sequence_length: 64
    summary_freq: 1000
    tau: 0.005
    use_recurrent: false
    vis_encode_type: simple
    reward_signals:
        extrinsic:
            strength: 1.0
            gamma: 0.99

Chameleon:
    vis_encode_type: nature_cnn
    init_entcoef: 0.25
    buffer_init_steps: 5000
    reward_signals:
        curiosity:
            strength: 0.02
            gamma: 0.99
            encoding_size: 256
```

https://github.com/afewvowels/unity_00_09_chameleon

**Comment by albertoxamin** [comment]
In the end I managed to make it work using SAC, it was a bug in the code that delayed the calculations of one frame.

However, why are you using such a low initial entropy?

**Comment by afewvowels** [comment]
Hm, I'll have to look into that.

I am using a low initial entropy because I'm using a discrete action space. The [docs](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Training-SAC.md) suggest an initial entropy coefficient of `0.05` - `0.5` for discrete.
